jit-service/src/main/java/org/dcsa/jit/service/TimestampRoutingService.java

```diff
@@ -0,0 +1,53 @@
+package org.dcsa.jit.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.dcsa.jit.persistence.entity.MessageRoutingRule;
+import org.dcsa.jit.persistence.entity.OutboxMessage;
+import org.dcsa.jit.persistence.repository.MessageRoutingRuleRepository;
+import org.dcsa.jit.persistence.repository.OutboxMessageRepository;
+import org.dcsa.jit.transferobjects.TimestampTO;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class TimestampRoutingService {
+  private final MessageRoutingRuleRepository messageRoutingRuleRepository;
+  private final OutboxMessageRepository outboxMessageRepository;
+  private final ObjectMapper objectMapper;
+
+  @Transactional
+  @SneakyThrows // JsonProcessingException when serializing timestamp
+  public void routeMessage(TimestampTO timestamp) {
+    String vesselIMONumber;
+    if (timestamp.vessel() != null && timestamp.vessel().vesselIMONumber() != null) {
+      vesselIMONumber = timestamp.vessel().vesselIMONumber(); // jit 1.2
+    } else {
+      vesselIMONumber = timestamp.vesselIMONumber(); // pre jit 1.2
+    }
+
+    List<MessageRoutingRule> messageRoutingRules = messageRoutingRuleRepository.findByVesselIMONumber(vesselIMONumber);
+    if (!messageRoutingRules.isEmpty()) {
+      String payload = objectMapper.writeValueAsString(timestamp);
+      List<OutboxMessage> outboxMessages = messageRoutingRules.stream()
+        .map(rule -> toOutboxMessage(rule, payload))
+        .toList();
+      outboxMessageRepository.saveAll(outboxMessages);
+    } else {
+      log.debug("No message routing rules found for vesselIMONumber '{}'", vesselIMONumber);
+    }
```

Comment:
What about all the message routing rules where `vessel_imo_number` is null? They should also match.

Reply:
Handled in the reposity with "OR vessselIMONumber IS NULL"

Reply:
With a method deceptively named like an automatic query, but it really used a custom one! :P

Reply:
Point taken, method name changed